From c3f2f3ed9f371b01eafae2583ec03abfd859336b Mon Sep 17 00:00:00 2001
From: lambdadroid <lambdadroid@gmail.com>
Date: Wed, 31 May 2017 11:50:07 +0200
Subject: [PATCH 02/25] ME176C: drivers/Makefile: Move mfd/ before gpu/

For some platforms, i915 may depend on the system's PMIC to do
backlight control. However, the PMIC driver in mfd/ is usually
loaded after gpu/, resulting in errors like:

*ERROR* Failed to own gpio for panel control
*ERROR* Failed to own the pwm chip

with backlight control broken.

Re-order the Makefile slightly to avoid this problem.
---
 drivers/Makefile | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/Makefile b/drivers/Makefile
index 5f5ccdbad21a..582d1a234902 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -61,6 +61,9 @@ obj-y				+= char/
 # iommu/ comes before gpu as gpu are using iommu controllers
 obj-$(CONFIG_IOMMU_SUPPORT)	+= iommu/
 
+# mfd/ comes before gpu/ as i915 depends on the PMIC for backlight control
+obj-y				+= mfd/
+
 # gpu/ comes after char for AGP vs DRM startup and after iommu
 obj-y				+= gpu/
 
@@ -72,7 +75,7 @@ obj-$(CONFIG_FB_INTEL)          += video/fbdev/intelfb/
 
 obj-$(CONFIG_PARPORT)		+= parport/
 obj-$(CONFIG_NVM)		+= lightnvm/
-obj-y				+= base/ block/ misc/ mfd/ nfc/
+obj-y				+= base/ block/ misc/ nfc/
 obj-$(CONFIG_LIBNVDIMM)		+= nvdimm/
 obj-$(CONFIG_DAX)		+= dax/
 obj-$(CONFIG_DMA_SHARED_BUFFER) += dma-buf/
-- 
2.17.0

